### DISCUSSION TITLE
Disable actionable buttons until the actual action can be consumed

### DISCUSSION THREAD

**Discussion by x80486** [ente-io/ente#493] [Enhancements]
## Description

This is something that, while a minor detail, could enhance the user experience. Basically, I think that disabling the buttons until the associated action(s) cannot be consumed provide more visual feedback for the users, regardless of their actions within the screen in question.

This enhancement would prevent user confusion and accidental actions, improving the overall usability of the application.

## Describe the Solution You Would Like

As an example: when editing a 2FA code, the `Save` button should only be enabled when the code actually changed, compared to the previous value. At the same time, the `Cancel` button (if any) should always be enabled.

## Describe Alternatives You Have Considered

N/A

## Additional Context

N/A

**Comment by Marukos** [comment]
Can i try to solve that?

**Comment by vishnukvmd** [comment]
Go for it!
